src/RevProc.cc

```diff
@@ -1893,16 +1940,16 @@ bool RevProc::ClockTick( SST::Cycle_t currentCycle ){
     }
 
     rtn = true;
-  }else{
+  } else {
     // wait until the counter has been decremented
     // note that this will continue to occur until the counter is drained
     // and the HART is halted
     output->verbose(CALL_INFO, 9, 0,
-                    "Core %u ; No available thread to exec PC= 0x%" PRIx64 "\n",
-                    id, ExecPC);
+                    "Core %" PRIu16 " ; Hart %" PRIu16 "; No available hart to exec PC= 0x%" PRIx64 "\n",
+                    id, HartToDecode, ExecPC);
     rtn = true;
     Stats.cyclesIdle_Total++;
-    if(HART_CTE.any()){
+    if( std::any_of(HART_CTE.begin(), HART_CTE.end(), [](bool x){ return true; })) {
```

Comment:
?

Seems like all of these could be replaced by:
```c++
if( !HART_CTE.empty() ){
```

Reply:
The number of entries in the vector doesn't go to 0... My intention with this `any_of` is to check if any of the elements inside of the vector are non-zero. I'm guessing by your confusion that's not what I'm doing 😂

Reply:
> The number of entries in the vector doesn't go to 0... My intention with this `any_of` is to check if any of the elements inside of the vector are non-zero. I'm guessing by your confusion that's not what I'm doing 😂

Correct, your predicate was always returning `true`, while it should have returned `x` (I think there is an `identity` function defined in one of the versions of C++, but I don't remember which one).

But switching to using `std::bitset` is simpler and faster.